packages/core/README.md

```diff
@@ -301,6 +301,296 @@ mcp.prompt("summarize", {
 });
 ```
 
+## Elicitation
+
+Elicitation enables MCP servers to request input from the client on behalf of the user during tool execution. This allows tools to gather additional information, confirm sensitive operations, or present choices to users through the connected AI application.
+
+### Usage Example
+
+```typescript
+import { z } from "zod";
+
+const DeleteRecordSchema = z.object({
+  recordId: z.string(),
+  tableName: z.string(),
+});
+
+mcp.tool("delete_database_record", {
+  description: "Delete a database record with user confirmation",
+  inputSchema: DeleteRecordSchema,
+  handler: async (args, ctx) => {
+    // Check if client supports elicitation
+    if (!ctx.client.supports("elicitation")) {
+      throw new Error("This tool requires a client that supports elicitation");
+    }
+
+    // Request user confirmation through elicitation
+    const response = await ctx.elicit({
+      type: "confirmation",
+      title: "Confirm Record Deletion",
+      description: `Are you sure you want to delete record "${args.recordId}" from table "${args.tableName}"? This action cannot be undone.`,
+      confirmationText: "Delete Record",
+      cancelText: "Cancel"
+    });
+
+    // Handle different response types
+    switch (response.type) {
+      case "accept":
+        // User confirmed - proceed with deletion
+        await deleteFromDatabase(args.tableName, args.recordId);
+        return {
+          content: [{ 
+            type: "text", 
+            text: `Record "${args.recordId}" has been deleted from "${args.tableName}".` 
+          }],
+        };
+      
+      case "decline":
+        return {
+          content: [{ 
+            type: "text", 
+            text: "Record deletion cancelled by user." 
+          }],
+        };
+      
+      case "cancel":
+        throw new Error("Operation was cancelled");
+      
+      default:
+        throw new Error("Unexpected elicitation response");
+    }
+  },
+});
+```
+
+### Setup Instructions
```

Comment:
this is quite a long section, might be worthwhile putting it into its own readme somewhere so as not to disrupt document (next H2 section is middleware, which has now been pushed down several hundred lines)